### ISSUE TITLE
catch ttl exceeded

### ISSUE THREAD

**Issue by vkalpias** [Automattic/kue#837]
Hello, 

Is there a proper way to know when a job has exceeded its ttl and it failed? All the following events I tried are not being fired when a job fails due to ttl:

### events on the job
* failed
* remove

### events on the queue
* job failed
* job remove

**Comment by behrad** [comment]
I don't think if I'm emitting `ttl exceeded`, Can you check the code and create a PR?

**Comment by vkalpias** [comment]
Shouldn't it fire a 'failed' event on the job / queue if it exceeds ttl?

**Comment by behrad** [comment]
Yes, and it currently is, however we may emit `ttl exceeded` also, If not, you can check error message in your failed event callback to check if it was a ttl exceeded string :)

**Comment by vkalpias** [comment]
I see, so you do fire a 'failed' event. I can't seem to be able to catch it.. Listening on the queue itself for "job failed" and also on the job itself for "failed" doesn't seem to catch that event.

**Comment by behrad** [comment]
if your job has >1 attempts, then `failed attempt` will be emitted until last attempt

**Comment by vkalpias** [comment]
Yeap tried that too still not getting fired. This is all assuming there is no problem in my code somewhere, although with the same code I know if a job fails or if a job is getting removed. I just don't get events when the job expires which makes me think it might be some sort of bug in kue itself.

**Comment by behrad** [comment]
Can you check inside the code?

**Comment by behrad** [comment]
any updates?

**Comment by vkalpias** [comment]
Haven't had time to fix this no

**Comment by behrad** [comment]
duplicate of #784

**Comment by johannds** [comment]
Any news on this? I'm setting TTL but the jobs don't get killed from the queue.

**Comment by orouji** [comment]
any updates?

**Comment by fetherolfjd** [comment]
With version 0.11.1 of Kue, I was unable to get any event for what should be a TTL failure. Turns out as @behrad mentioned, a ```ttl exceeded``` event is indeed being fired, and it seems the job is allowed to continue life through to any natural completion if no other actions are taken.

**Comment by RWOverdijk** [comment]
👍 this would be a useful feature.

**Comment by throrin19** [comment]
+1 this feature is so important for me

**Comment by pvraj** [comment]
Hello,
Is there any update on this? I also looked at [issue 784](https://github.com/Automattic/kue/issues/784).

I have jobs that fire a `ttl exceeded` event but do not emit a failure event. The `ttl exceeded` jobs still continue and are processed by queue (just like a normal job that did not exceed the ttl threshold), making the ttl exceeded somewhat useless for me (since I require the `ttl exceeded` jobs to fail immediately).
